src/System.Collections/tests/Performance/Perf.HashSet.cs

```diff
@@ -0,0 +1,501 @@
+    {
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+            int[] startingElements = intGenerator.MakeNewTs(initialSetSize);
+            int[] subsetToCheck = intGenerator.GenerateSelectionSubset(startingElements, countToCheck);
+            bool present;
+            HashSet<int> theSet = new HashSet<int>();
+            theSet.UnionWith(startingElements);
+
+            using (iteration.StartMeasurement())
+            {
+                foreach (int thing in subsetToCheck)
+                {
+                    present = theSet.Contains(thing);
+                }
+            }
+        }
+    }
+
+    [Benchmark]
+    [InlineData(InitialSetSize, 1)]
+    [InlineData(InitialSetSize, 100)]
+    [InlineData(InitialSetSize, 10000)]
+    public static void Remove(int initialSetSize, int countToRemove)
+    {
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+            int[] startingElements = intGenerator.MakeNewTs(initialSetSize);
+            int[] stuffToRemove = intGenerator.GenerateSelectionSubset(startingElements, countToRemove);
+            HashSet<int> theSet = new HashSet<int>();
+            theSet.UnionWith(startingElements);
+
+            using (iteration.StartMeasurement())
+            {
+                foreach (int thing in stuffToRemove)
+                {
+                    theSet.Remove(thing);
+                }
+            }
+        }
+    }
+
+    [Benchmark]
+    [InlineData(InitialSetSize_small)]
+    public static void Clear(int initialSetSize)
+    {
+        RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+        int[] startingSet = intGenerator.MakeNewTs(initialSetSize);
+        HashSet<int> theSet = new HashSet<int>();
+
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            theSet.UnionWith(startingSet);
+
+            using (iteration.StartMeasurement())
+                theSet.Clear();
+        }
+    }
+
+    [Benchmark]
+    [InlineData(MaxStartSize, InitialSetSize_small)]
+    public static void Union(int startSize, int countToUnion)
+    {
+        RandomTGenerator<int> intGenerator = new RandomTGenerator<int>(InstanceCreators.IntGenerator);
+        int[] startingElements = intGenerator.MakeNewTs(MaxStartSize);
+        int[] stuffToUnion = intGenerator.GenerateMixedSelection(startingElements, countToUnion);
+        HashSet<int> theSet = new HashSet<int>();
+
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            theSet.UnionWith(startingElements);
+
+            using (iteration.StartMeasurement())
+                theSet.UnionWith(stuffToUnion);
+
+            theSet.Clear();
```

Comment:
Should we have a version of the test that doesn't clear, so that we test the cost of union when there's nothing to add?